cmd/secrets-provider/main.go

```diff
@@ -186,6 +181,14 @@ func retryableSecretsProvider(
 		secretsConfig.RetryCountLimit,
 		provideSecrets,
 	)
+
+	provideSecrets = secrets.PeriodicSecretProvider(
+		secretsConfig.SecretsRefreshInterval,
+		getContainerMode(),
+		provideSecrets,
+		// Set a large timeout, basically forever
+		secretsConfigProvider.MaxRefreshTimeout,
```

Comment:
Is this MaxRefreshTimeout used for implementing a close proximity for "sleep forever" functionality? If so, I think we can do this differently, and have the SP block forever.